src/sotauptaneclient.cc

```diff
@@ -165,7 +153,7 @@ Json::Value SotaUptaneClient::AssembleManifest() {
     (refname += "unknown-") += hash;
   }
   Json::Value unsigned_ecu_version =
-      OstreePackage(refname, hash, "").toEcuVersion(uptane_repo.getPrimaryEcuSerial(), Json::nullValue);
+      OstreePackage(refname, hash, "").toEcuVersion(uptane_repo.getPrimaryEcuSerial(), operation_result);
 
   result.append(uptane_repo.getCurrentVersionManifests(unsigned_ecu_version));
```

Comment:
This also seems like a misnomer (even if it isn't your fault or related specifically to this PR!). It is just getting one manifest, namely for the primary. The name implies that it is getting multiple (presumably from the secondaries), but that is in fact done by this function (`AssembleManifest`). This is the only place that `getCurrentVersionManifests` is called.

Reply:
In fact `getCurrentVersionManifests` doesn't even get anything, it just signs its argument.